module/data/shared/source-field.mjs

```diff
@@ -38,8 +38,8 @@ export default class SourceField extends SchemaField {
    * @param {string} uuid  Compendium source or document UUID.
    */
   static prepareData(uuid) {
-    const pkg = SourceField.getPackage(uuid);
-    this.bookPlaceholder = SourceField.getModuleBook(pkg) ?? "";
+    const { pkg, flags } = SourceField.getPackage(uuid) ?? {};
```

Comment:
Let's keep `getPackage` returning a package only. We can `parseUuid` here instead in order to retrieve the collection flags, and then modify `getPackage` to accept a `CompendiumCollection` alternatively in order to preserve its API and avoid doing the work twice.

```js
// prepareData
const { collection } = foundry.utils.parseUuid(uuid) ?? {};
const flags = collection?.metadata?.flags;
const pkg = SourceField.getPackage(collection);

// getPackage
if ( !uuidOrCollection ) return null;
const pack = typeof uuidOrCollection === "string" ? foundry.utils.parseUuid(uuidOrCollection)?.collection?.metadata : uuidOrCollection.metadata;
```